Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -680,15 +680,17 @@ namespace Babylon
 
         bgfx::VertexLayout vertexLayout{};
         vertexLayout.begin();
-        const bgfx::Attrib::Enum attrib = static_cast<bgfx::Attrib::Enum>(location);
+
+        const auto locationIndex = m_currentProgram->VertexAttributeLocations[kind];
```

Comment:
I don't understand this change. Why is this necessary? The js side gets the location ultimately via `NativeEngine::GetAttributes` which is all on the native side of things. What does using strings for kind do?

Reply:
I agree, I don't think this (or my other js changes) are necessary either, since we are already given the proper vertex location. I think currently the order in which we put elements in `vertexArray.vertexBuffers` is causing my current issues with data ordering

Reply:
I removed this additional lookup and reverted my changes to the babylon.js contract, instead I believe the real issue is how we're passing vertex arrays from our babylon representation to bgfx's, i opted to change our datastructure for representing which array maps to a given attribute location so that we can insert these elements in an arbitrary order.